source/beamline/BioXAS/BioXASBeamlineComponent.cpp

```diff
@@ -11,9 +12,42 @@ BioXASBeamlineComponent::~BioXASBeamlineComponent()
 
 }
 
+bool BioXASBeamlineComponent::stop()
+{
+	// Want to return true if all the controls are valid and those
+	// that can be stopped are stopped, false otherwise. Returns
+	// true if there are no child controls, the expectation is that
+	// this control can't be moving otherwise.
+
+	bool childrenValid = true;
+	bool childrenStopped = true;
+
+	foreach (AMControl *control, children_) {
```

Comment:
Can `children_` contain a null pointer?  If so, that would be a bit troubling since you could have a situation where `childrenValid` was false, but `childrenStopped` was true.